### PR TITLE
[#9079] Set up Angular 6 project

### LINKED ISSUE #9079: Set up foundation for front-end migration project
To kick off [this project](https://github.com/TEAMMATES/teammates/projects/4), some foundations need to be laid out first:
- [x] Step 1: Set up Angular 6 front-end project
  - Included: lint/build/test routine (including reliable code coverage), separate dev server from the back end, hot reload in development mode, source maps, minification when building for production usage
- [x] Step 2: Set up REST-ful back end servlet, ~possibly with the help of [Jersey](https://medium.com/@tejohnso/a-minimal-java-api-with-jetty-jersey-jackson-and-gradle-9842a145815e),~ and user authentication workflow
- [x] Step 3: Migrate few pages; concurrently, set up example of how migration for each page should be done systematically.

- [ ] Step 4 (can be deferred): Set up example tests, ranging from front-end component tests (with Karma/Jasmine), API endpoint tests (with TestNG), and E2E tests (with Selenium).

### PR DESCRIPTION
**Pull request by wkurniawan07**
Step 1 of #9079. 

Changes:
- Set up Angular 6 project with complete lint/build/test routine
- Removed webpack config and all references to JavaScript (and by extension, ESLint) and JSP
- Separated CI tests to component tests (separate for back-end and front-end) and E2E tests, and removed "Local tests"
- Upgraded Selenium to `3.x`; this is trivial right now because all existing UI tests are deactivated; problems will crop up later on

List of commands to try:
- `npm run start`: Starts local dev server at port `4200` in watch mode with hot reload available
- `npm run lint:ts`: Lints all TS files; replaces `npm run lint:js`
- `npm run build`: Builds front-end files for production usage
- `npm run test`: Runs all tests in watch mode
- `npm run coverage`: Runs all tests once and produces code coverage report

Caveats:
- The visual display is very disorganized right now, although all the click/... interactions are there; I have not had the time to tidy them up
- I have not examined the SEO impact (SPAs are infamous for being SEO-unfriendly)
- Also I have not tried to optimize development experience in IntelliJ because most of the front-end plugins need Ultimate version

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@xpdavid would be great if you can help review. I'll walk through the code if you need to. Also if you have time, you may want to see if this setup works in IntelliJ 😛

File: docs/glossary.md [hunk 1]
```diff
@@ -46,13 +46,13 @@ This document lines out the common terms used in the project.
 `* tests`:
 
 + **All tests**: The full test suite.
-+ **CI tests**: Tests to be run on the CI server.
-+ **Local tests**: Tests to be run locally by developers.
++ **Component tests**: Unit and integration tests, i.e. white-box tests. There are separate component test suites for front-end and back-end.
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply docs/glossary.md]: `npm run test` vs `./gradlew componentTests`

File: src/test/resources/test.template.properties [hunk 2]
```diff
@@ -26,6 +26,12 @@ test.selenium.browser=firefox
 # e.g test.firefox.path=/Applications/Firefox.app/Contents/MacOS/firefox-bin
 test.firefox.path=
 
+# This is the path for Selenium's Geckodriver executable.
+# It is required if you are running test with Firefox.
+# e.g test.chromedriver.path=C:/Users/YourName/Downloads/geckodriver.exe
+# e.g test.chromedriver.path=/Users/YourName/Downloads/geckodriver
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/test/resources/test.template.properties]: Good point. I actually haven't updated `development.md` because there are quite a lot of changes.

**Comment by wkurniawan07** [comment]
> Admittedly, I only have a little background about Angular 6.

Nothing to worry about. Almost nobody needs to touch about 80% of the code I've added (and to be frank most are auto-generated by Angular CLI). Once I set up enough service classes and enough sample boilerplate codes, the learning curve should be rather gentle.

<sub>Of course it will be another story later on e.g. when we port a library to its Angular-equivalent 😛</sub>

**Comment by wkurniawan07** [comment]
> As for the setup in Intellij, it is quite easy as you just need to click enable when Angular framework is detected. In addition, one should also disable `eslint` and enable `tslint` in the setting.

@xpdavid can help me create a short write-up, or a commit on how you would change the relevant section in `ide-setup.md`?

**Comment by xpdavid** [comment]
> can help me create a short write-up, or a commit on how you would change the relevant section in ide-setup.md?

Let me import the project from scratch in order to reproduce the message.

**Comment by wkurniawan07** [comment]
> I guess you don't mean to set up a `npm` run configuration in the IDE right?

I actually was referring to this, but I guess this is an unnecessary overhead and devs can just run any npm script on its own? If that is the case, this part can be skipped.
Background: I was asking that because Eclipse's Angular IDE plugin creates shortcut to run dev server, much like Eclipse's GCP plugin.

**Comment by xpdavid** [comment]
Here is the configuration for generic npm task setup. It is just a shortcut of running `npm start`.
0fd44d225d4f69f12765f058e40cb018d79b04f2

**Comment by wkurniawan07** [comment]
@xpdavid I decided not to include any extra information (I even removed the Eclipse Angular IDE information); let the npm scripts be CLI-exclusive. If anyone wants to create an IDE integration, they're free to do it on their own.

File: package.json [hunk 3]
```diff
@@ -1,44 +1,63 @@
...
+    "@angular/common": "^6.1.4",
+    "@angular/compiler": "^6.1.4",
+    "@angular/core": "^6.1.4",
+    "@angular/forms": "^6.1.4",
+    "@angular/http": "^6.1.4",
+    "@angular/platform-browser": "^6.1.4",
+    "@angular/platform-browser-dynamic": "^6.1.4",
+    "@angular/router": "^6.1.4",
+    "@ng-bootstrap/ng-bootstrap": "^3.1.0",
+    "bootstrap": "^4.1.3",
```
Comment (xpdavid) [inline package.json]: Just realize we are in Bootstrp 4 🎉

Comment (wkurniawan07) [reply package.json]: Well this is precisely why the UI is very disorganized right now. We'll need a serious work to get the UI back up to standard, but that's not my highest priority right now.

**Pull request merged**